batch_object_tagger.tf

```diff
@@ -39,6 +39,7 @@ data "aws_iam_policy_document" "pdm_object_tagger" {
 
     resources = [
       "${data.terraform_remote_state.common.outputs.config_bucket.arn}/*",
+      "${data.terraform_remote_state.common.outputs.published_bucket.arn}/*"
```

Comment:
you need to split this statement, one for the csv access and one for the published data access. they require different permissions. You also need a 3rd statement for the configs KMS key (output from common)